src/Symfony/Component/Form/Util/StringUtil.php

```diff
@@ -33,7 +33,7 @@ private function __construct()
      */
     public static function trim($string)
     {
-        if (null !== $result = @preg_replace('/^[\pZ\pC]+|[\pZ\pC]+$/u', '', $string)) {
+        if (null !== $result = @preg_replace('/^[\pZ\p{Cc}\p{Cf}\p{Co}\p{Cs}]+|[\pZ\p{Cc}\p{Cf}\p{Co}\p{Cs}]+$/u', '', $string)) {
```

Comment:
Made some research and stumbled over this: https://www.pcre.org/original/doc/html/pcresyntax.html#SEC5
Using `\pC` is too general imo. I replaced it by all of this group except the unassigned category.
This should be a proper solution for all PHP/PCRE version to be more future proof or do you see a problem with this?
I am not even sure if `Co` and `Cs` should or have to be included.